src/app/ApplicationTemplate.Tests.ApiClient/ApiEndpointTestBase.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly, DisableTestParallelization = true)]
+
+namespace ApplicationTemplate.Tests;
+
+/// <summary>
+///  Gives access to the API endpoints and their configuration.
+/// </summary>
+public abstract class ApiEndpointTestBase : IDisposable
+{
+	private readonly IHost _host;
+
+	private bool _disposedValue;
+
+	protected ApiEndpointTestBase()
+	{
+		_host = new HostBuilder()
+			.AddConfiguration(string.Empty, new TestEnvironmentManager())
+			.ConfigureServices((context, s) => s
+				.AddDiagnostics(context.Configuration)
+				.AddMock(context.Configuration)
+				.AddApi(context.Configuration)
+				.AddSerialization()
+		).Build();
+	}
+
+	public void Dispose()
+	{
+		// Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method.
```

Comment:
Why not just call _host.Dispose(); directly?

Reply:
I just used the VS implementation of the dispose pattern (using the light bulb). I think it looks clean but properly overkill for such a small class.

Reply:
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1063

Reply:
Fixed (I suppressed it).